src/Request.ts

```diff
@@ -178,3 +170,61 @@ export function isRetryable (status?: number): boolean {
       ].indexOf(status) !== -1)
     )
 }
+
+export function fetch(endpoint: string): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    const parsedUrl = url.parse(endpoint)
+
+    const req = (parsedUrl.protocol === 'https:' ? https : http).get(endpoint, res => {
+      res.pipe(concat((bodyBuffer: Buffer) => {
+        if (res.statusCode === 200) {
+          return resolve(bodyBuffer.toString())
+        }
+
+        const err = new NetworkError(`HTTP status ${res.statusCode} received from bundle server`)
+        err.responseText = bodyBuffer.toString()
+
+        if (!isRetryable(res.statusCode)) {
+          err.isRetryable = false
+        }
+
+        if (res.statusCode && (res.statusCode >= 400 && res.statusCode < 500)) {
+          err.code = NetworkErrorCode.MISC_BAD_REQUEST
+        } else {
+          err.code = NetworkErrorCode.SERVER_ERROR
+        }
+
+        return reject(err)
+      }))
+    })
+
+    addErrorHandler(req, reject)
+    addTimeout(req, reject)
+  })
+}
+
+function addErrorHandler(req: http.ClientRequest, reject: (reason: NetworkError) => void): void {
+  req.on('error', e => {
+    const err = new NetworkError('Unknown connection error')
+    err.cause = e
+
+    const failureReason = (e as any).code
```

Comment:
There's a lint warning on this line – I think the lint rules would prefer this to be cast `as unknown`, or to speculatively cast it as the actual type that it would be, you could `NodeJS.ErrnoException`.

Reply:
Is this actually the best way to do this? I couldn't find any way to make TS let me check if this property existed, but I feel like that's a nicer solution than casting to a type that it may not actually be

Reply:
you could cast it or type the function argument as the union `Error | NodeJS.ErrnoException`? Though that might be a little weird since [`ErrnoException` extends `Error`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f621a2d639f01584eb88975c182a61a52d4d7a8b/types/node/v10/globals.d.ts#L546-L552)…

otherwise `as unknown` is probably sufficient to let you check the existence of the property, without it shouting at you for using `any`